### PR TITLE
[SPARK-25384][SQL] Removing of spark.sql.fromJsonForceNullableSchema

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

In the PR, I propose to remove the `spark.sql.fromJsonForceNullableSchema` flag since disabling it can cause corrupted output. The flag was introduced only for backward compatibility in minor versions. The PR targets Spark 3.0 in which the flag can be removed.

## How was this patch tested?

It was tested by `JsonExpressionsSuite`, `JsonFunctionsSuite` and `JsonSuite`

### CODE REVIEW

**Comment by kiszk** [comment]
Is it better to add a description to `docs/sql-programming-guide.md`?

**Comment by HyukjinKwon** [comment]
Yea, let's note so that we can track what we change. cc @gatorsmile as well

**Comment by MaxGekk** [comment]
> Is it better to add a description to docs/sql-programming-guide.md?
> Yea, let's note so that we can track what we change.

@kiszk @HyukjinKwon May I ask you to clarify this. Do you want a description after removing the flag or instead of removing just to describe a behavior when the flag is disabled in `docs/sql-programming-guide.md`?

**Comment by kiszk** [comment]
I think that it is good to describe the flag is removed. This is because the default behavior is not changed and users can know the flag is not effective in the next release.
cc @HyukjinKwon @gatorsmile

**Comment by dongjoon-hyun** [comment]
@MaxGekk . If this targets for 3.0, can we postpone this until the master branch get `3.0.0-SNAPSHOT`?
> The PR targets Spark 3.0 in which the flag can be removed.

**Comment by MaxGekk** [comment]
>  If this targets for 3.0, can we postpone this until the master branch get 3.0.0-SNAPSHOT?

@dongjoon-hyun Yes, sure.

**Comment by MaxGekk** [comment]
@dongjoon-hyun The master branch became `2.5.0-SNAPSHOT` recently. Can we move forward with this PR?

**Comment by dongjoon-hyun** [comment]
Sorry, @MaxGekk . It looks opposite to me. It's fixed at 2.3.1 and 2.4.0. `2.5.0-SNAPSHOT` means we cannot do this for now. Only `3.0.0` will allow this.

How do you think about this, @gatorsmile ?

**Comment by MaxGekk** [comment]
> Only 3.0.0 will allow this.

@dongjoon-hyun I keep this PR open for a while. Maybe `2.5.0-SNAPSHOT` will become `3.0.0-SNAPSHOT` in the near future.

**Comment by dongjoon-hyun** [comment]
Sure, +1 for keeping this PR.

**Comment by MaxGekk** [comment]
@dongjoon-hyun If you don't mind, I would create a separate PR and deprecate the config in Spark 2.5 before removing it in Spark 3.0.

**Comment by MaxGekk** [comment]
I am going to close the PR since I don't see any reasons so far to maintain it up to Spark 3.0.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala [hunk 1]
```diff
@@ -517,12 +517,10 @@ case class JsonToStructs(
     timeZoneId: Option[String] = None)
   extends UnaryExpression with TimeZoneAwareExpression with CodegenFallback with ExpectsInputTypes {
 
-  val forceNullableSchema = SQLConf.get.getConf(SQLConf.FROM_JSON_FORCE_NULLABLE_SCHEMA)
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala]: When we reopened this PR, please do not forget to remove SQLConf.FROM_JSON_FORCE_NULLABLE_SCHEMA from SQLConf.scala

**Pull request closed without merging**